builder/osc/bsusurrogate/step_register_omi.go

```diff
@@ -4,86 +4,81 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/antihax/optional"
 	"github.com/hashicorp/packer-plugin-sdk/multistep"
 	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
-	"github.com/outscale/osc-sdk-go/osc"
+	"github.com/outscale/osc-sdk-go/v2"
+	oscgo "github.com/outscale/osc-sdk-go/v2"
 	osccommon "github.com/outscale/packer-plugin-outscale/builder/osc/common"
 )
 
 // StepRegisterOMI creates the OMI.
 type StepRegisterOMI struct {
 	RootDevice    RootBlockDevice
-	OMIDevices    []osc.BlockDeviceMappingImage
-	LaunchDevices []osc.BlockDeviceMappingVmCreation
-	image         *osc.Image
+	OMIDevices    []oscgo.BlockDeviceMappingImage
+	LaunchDevices []oscgo.BlockDeviceMappingVmCreation
+	image         *oscgo.Image
 	RawRegion     string
 }
 
 func (s *StepRegisterOMI) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
-	oscconn := state.Get("osc").(*osc.APIClient)
+	oscconn := state.Get("osc").(*(osccommon.OscClient))
 	snapshotIds := state.Get("snapshot_ids").(map[string]string)
 	ui := state.Get("ui").(packersdk.Ui)
 
 	ui.Say("Registering the OMI...")
 
 	blockDevices := s.combineDevices(snapshotIds)
-
-	registerOpts := osc.CreateImageRequest{
-		ImageName:           config.OMIName,
-		Architecture:        "x86_64",
-		RootDeviceName:      s.RootDevice.DeviceName,
-		BlockDeviceMappings: blockDevices,
+	architecture := "x86_64"
+	registerOpts := oscgo.CreateImageRequest{
+		ImageName:           &config.OMIName,
+		Architecture:        &architecture,
```

Comment:
I've discovered that the SDK have a Utils package which contains that:
```func PtrString(v string) *string { return &v }```(see [link](https://github.com/outscale/osc-sdk-go/blob/7a11a0906c025139fb2b6f25cbd8f38890121333/v2/utils.go#L38)).

I do not ask for change, just to let you know